System/RTETools.h

```diff
@@ -38,37 +40,59 @@ namespace RTE {
 
 #pragma region Random Numbers
 	/// <summary>
-	/// Seeds the rand with the current runtime time.
+	/// Seed the mt19937 random number generator. mt19937 is the standard mersenne_twister_engine.
+	/// </summary>
+	void SeedRNG();
+
+	/// <summary>
+	/// Seed the mt19937 random number generator. mt19937 is the standard mersenne_twister_engine.
 	/// </summary>
-	void SeedRand();
+	/// <param name="seed">Seed for the random number generator.</param>
+	void SeedRNG(unsigned int seed);
 
 	/// <summary>
-	/// A good rand function that return a float between 0.0 and 0.999.
+	/// Uniformly distributed random double in the range [0, 1].
 	/// </summary>
-	/// <returns>Random number.</returns>
+	/// <returns>Uniformly distributed random double in the range [0, 1].</returns>
 	double PosRand();
 
 	/// <summary>
-	/// A good rand function that returns a floating point value between -1.0 and 1.0, both inclusive.
+	/// Uniformly distributed random float in the range [-1, 1].
 	/// </summary>
-	/// <returns>Random number.</returns>
-	double NormalRand();
+	/// <returns>Uniformly distributed random float in the range [-1, 1].</returns>
+	float NormalRand();
 
 	/// <summary>
-	/// A good rand function that returns a floating point value between two given thresholds, the min being inclusive, but the max not.
+	/// Template function which returns a Uniformly distributed random number in the range [0, 1].
 	/// </summary>
-	/// <param name="min">Minimum value this can return.</param>
-	/// <param name="max">Maximum value this can return.</param>
-	/// <returns>Random number between limits.</returns>
-	double RangeRand(float min, float max);
+	/// <returns>Uniformly distributed random number in the range [0, 1].</returns>
+	template <typename floatType = float>
+	typename std::enable_if<std::is_floating_point<floatType>::value, floatType>::type RandomNum() {
+		return std::uniform_real_distribution<floatType>(floatType(0.0), std::nextafter(floatType(1.0), std::numeric_limits<floatType>::max()))(g_RNG);
+	}
+
+	template <typename intType>
```

Comment:
Can you put method comments on the int versions too? I know they'll be 99% the same, but autogenerated docs and vs code hints won't be able to figure out to apply the comments to this method too, so we have to put them everywhere.

Reply:
Yeah I'll add the comments.

As I understand it, the int template block is a "template specialization", so the float and int blocks together constitute a single "function template" (using a technique called SFINAE).
In this case I would assume VS hints would recognize that, but that is not the case unfortunately.